File: src/test/java/cspom/dimacs/ParserTest.java
package cspom.dimacs;

import java.io.IOException;

import org.junit.Test;

import cspom.CSPOM;
import cspom.CSPParseException;

public class ParserTest {

	private static final String FILENAME = "flat30-1.cnf";

	@Test
	public void test() throws IOException, CSPParseException {
		final CSPOM cspom = new CSPOM();
		final CNFParser parser = new CNFParser(cspom);
		parser.parse(ParserTest.class.getResource(FILENAME).openStream());
		System.out.println(cspom);
	}

}

File: src/test/scala/cspom/dimacs/ParserTest.scala
package cspom.dimacs;


import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

final class ParserTest extends AnyFlatSpec with Matchers {

  val FILENAME = "flat30-1.cnf";

  "CNFParser" should s"parse $FILENAME" in {
    val cspom = CNFParser(classOf[ParserTest].getResourceAsStream(FILENAME)).get

    cspom.expressionMap.count shouldBe 90
    cspom.constraints should have size 300

  }

}
